### PR TITLE
[New Rule] Remote Desktop Enabled in Windows Firewall

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/367

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[metadata]
+creation_date = "2020/10/13"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies use of the netsh.exe utilty to enable inbound RDP in the Windows Local Firewall."
```
Comment (bm11100) [inline rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml:9]:
```suggestion
description = "Identifies use of the network shell utility (netsh.exe) to enable inbound Remote Desktop Protocol (RDP) connections in the Windows Local Firewall."
```

File: rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[metadata]
+creation_date = "2020/10/13"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = "Identifies use of the network shell utility (netsh.exe) to enable inbound Remote Desktop Protocol (RDP) connections in the Windows Local Firewall."
```
Comment (brokensound77) [inline rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml:9]: needs lint

File: rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+rule_id = "074464f9-f30d-4029-8c03-0ed237fffec7"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+  (process.name == "netsh.exe" or process.pe.original_file_name == "netsh.exe") and
+  wildcard(process.args,"localport=3389","RemoteDesktop", "group=\"remote desktop\"") and 
+  wildcard(process.args, "action=allow", "enable=Yes", "enable")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml:25]:
```suggestion
process where event.type in ("start", "process_started") and
  (process.name : "netsh.exe" or process.pe.original_file_name == "netsh.exe") and
  process.args : ("localport=3389", "RemoteDesktop", "group=\"remote desktop\"") and 
  process.args : ("action=allow", "enable=Yes", "enable")
```

**Comment by Samirbous** [comment]
@brokensound77 not sure what caused this error ? 

image

**Comment by dstepanic17** [comment]
@Samirbous Hey, I tried the query out and ran into the following error: 

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parsing_exception",
        "reason" : "line 4:34: mismatched input ',' expecting {'and', 'in', 'not', 'or', '+', '-', '*', '/', '%', ')'}"
      }
    ],
    "type" : "parsing_exception",
    "reason" : "line 4:34: mismatched input ',' expecting {'and', 'in', 'not', 'or', '+', '-', '*', '/', '%', ')'}",
    "caused_by" : {
      "type" : "input_mismatch_exception",
      "reason" : null
    }
  },
  "status" : 400
}
```

image

**Comment by Samirbous** [comment]
> @Samirbous Hey, I tried the query out and ran into the following error:
> 
> ```
> {
>   "error" : {
>     "root_cause" : [
>       {
>         "type" : "parsing_exception",
>         "reason" : "line 4:34: mismatched input ',' expecting {'and', 'in', 'not', 'or', '+', '-', '*', '/', '%', ')'}"
>       }
>     ],
>     "type" : "parsing_exception",
>     "reason" : "line 4:34: mismatched input ',' expecting {'and', 'in', 'not', 'or', '+', '-', '*', '/', '%', ')'}",
>     "caused_by" : {
>       "type" : "input_mismatch_exception", 
>       "reason" : null
>     }
>   },
>   "status" : 400
> }
> ```
> 
> image

Hey @dstepanic17 I know, the new EQL syntax (@brokensound77 can confirm) something : ("a", "b", "c") or something : "a*b*c" are not supported in 7.10, I've tested on 7.10 with initial PR content, have no guarantees it will work as expected on 7.11.

**Review by dstepanic17 — APPROVED** [review]
LGTM

File: rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+[metadata]
+creation_date = "2020/10/13"
+maturity = "production"
+updated_date = "2020/10/13"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies use of the network shell utility (netsh.exe) to enable inbound Remote Desktop Protocol (RDP) connections in
+the Windows Local Firewall.
```
Comment (bm11100) [inline rules/windows/defense_evasion_enable_inbound_rdp_with_netsh.toml:10]:
```suggestion
Identifies use of the network shell utility (netsh.exe) to enable inbound Remote Desktop Protocol (RDP) connections in
the Windows Firewall.
```

**Review by bm11100 — APPROVED** [review]

**Pull request merged**